### PR TITLE
[ISSUE #3116]Refactor ExtensionClassLoader implement

### LINKED ISSUE #3116: [Enhancement] Refactor ExtensionClassLoader implement
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Refactor ExtensionClassLoader implement

### Describe the solution you'd like

Refactor ExtensionClassLoader implement and add some test cases

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mxsm**
Fixes #3116 

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

-  *Refactor ExtensionClassLoader implement*

-  *add some test cases*
-  *polish code style*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java [hunk 1]
```diff
@@ -50,33 +50,34 @@ public class JarExtensionClassLoader implements ExtensionClassLoader {
...
-            new ConcurrentHashMap<>(16);
+    private static final String EVENTMESH_EXTENSION_PLUGIN_DIR = System.getProperty(EVENT_MESH_PLUGIN_DIR,
+        Joiner.on(File.separator).join(Lists.newArrayList(".", "plugin")));
 
-    private static final String EVENTMESH_EXTENSION_PLUGIN_DIR =
-            System.getProperty(EVENT_MESH_PLUGIN_DIR,
-                    Joiner.on(File.separator).join(Lists.newArrayList(".", "plugin")));
+    private static final JarExtensionClassLoader INSTANCE = new JarExtensionClassLoader();
+
+    private final ConcurrentHashMap<Class<?>, Map<String, Class<?>>> extensionClassCache = new ConcurrentHashMap<>(16);
```
Comment (jonyangx) [inline eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java:58]: Why not use static modifier?

File: eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/MetaInfExtensionClassLoader.java [hunk 2]
```diff
@@ -38,12 +38,21 @@ public class MetaInfExtensionClassLoader implements ExtensionClassLoader {
 
     private static final Logger logger = LoggerFactory.getLogger(MetaInfExtensionClassLoader.class);
```
Comment (jonyangx) [inline eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/MetaInfExtensionClassLoader.java:39]: Suggest use lombok slf4j annotation.

File: eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java [hunk 3]
```diff
@@ -38,4 +38,18 @@ public void testGetPrototypeExtension() {
...
+        ExtensionException exception = null;
+        try {
+            EventMeshExtensionFactory.getExtension(null, "eventmesh");
+        } catch (Exception ex) {
+            if (ex instanceof ExtensionException) {
+                exception = (ExtensionException) ex;
+            }
+        }
+        Assert.assertNotNull(exception);
+        EventMeshExtensionFactory.getExtension(TestPrototypeExtension.class, null);
```
Comment (jonyangx) [inline eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java:53]: Suggest add asserttion.

File: eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java (hunk 1 above)
Comment (mxsm) [reply eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java:58]: @jonyangx JarExtensionClassLoader is a singleton mode, and extensionClassCache is a property

**Review by jonyangx — APPROVED** [review]

**Pull request merged**